### DISCUSSION TITLE
HTML output: is there a native way to add a horizontal scrollbar for very large markdown table?

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#2983] [Q&A] [answered]
Hi,
I have a large table: is there a native way to add a horizontal scrollbar for very large markdown table?

Thank you

![image](https://user-images.githubusercontent.com/30607/197353301-5cf78df7-bbb5-4fc5-98b0-749ce319334d.png)

**Comment by mcanouil** [comment]
Hi,

yes, using HTML/CSS.
For a more detailed answer, could you provide a small working reproducible example.

**Reply by aborruso** [reply]
@mcanouil first of all thank you very much for what you do to share great Quarto contents. You are often my lighthouse in the night!

I know, I can do all with HTML and CSS, also not use quarto :)
But I explained myself badly.

As "native" I wanted to mean: is there already something ready? In example some CSS class to apply (i.e. `.table-wrapper`)??

Moreover the default classes of tables generated by code, they seem better to me, the tables are more beautiful and readable. They [use](https://github.com/quarto-dev/quarto-cli/commit/0a352d65b52ff1bd2a8dd33f1a63f876cfa2b0ae)  `table-sm` and `table-striped` classes.

Moreover I have tried to use `:::{.table-sm .table-striped}`  for my markdown table, ma this seems not to work.

Then there is not a ready way to do it. I think it would be convenient to have a class to apply horizontal and vertical scrolling (after a limit of rows).

For my table I'm using this

```css
.table-wrapper {
  overflow-x: scroll;
  font-size: 13px;
}

.even {background: #f2f2f2}
.odd {background: #FFF}
```

Thank you

![image](https://user-images.githubusercontent.com/30607/197355983-cf8f0833-8957-4cc8-883e-1f62007e34e9.png)

**Reply by mcanouil** [reply]
You are welcome and than your for this feedback/compliment! 😉

Since Quarto uses Bootstrap, everything in it (or almost) is available to you.
On top of that there are some `.quarto-*` CSS classes, but not sure those will help you.
There some "overflow" classes: <https://quarto.org/docs/authoring/article-layout.html#overflowing-content>

To note for "epub" format, Quarto defines:

```css
.cell-output-display {
  overflow-x: scroll;
}
```

**Comment by cderv** [comment, marked as answer]
Leveraging Bootstrap feature getbootstrap.com/docs/4.0/content/tables/#responsive-tables you can wrap your table in a div with `.table-responsive` class. 

Example at : https://github.com/quarto-dev/quarto-cli/issues/4942#issuecomment-1482691149

**Reply by aborruso** [reply]
Thank you @cderv I know because I had also opened an [issue](https://github.com/quarto-dev/quarto-cli/issues/2997) about it, which was excellently resolved.

**Reply by cderv** [reply]
Awesome ! thanks for the back references!
